src/otx/algo/visual_prompting/zero_shot_segment_anything.py

```diff
@@ -594,8 +596,8 @@ def _decide_cascade_results(
                 # all predicted masks were zero masks, ignore them.
                 return None, torch.zeros(masks.shape[-2:], device="cpu")
 
-            best_idx = torch.argmax(scores[0])
-        return logits[:, best_idx], masks[0, best_idx]
+            best_idx = int(torch.argmax(scores[0]))
```

Comment:
Converting to Python integer is necessary? It is not desired because data transfer between devices may be required for this conversion.

Reply:
As you said, it is unnecessary operation, but it is intended to match type with `best_idx` which can be assigned at L584.
If you think it is unnecessary, I'll revert it.

Reply:
Please revert it

Reply:
Done at https://github.com/openvinotoolkit/training_extensions/pull/3090/commits/918a22628539123fd12b905d83873268c3e42356.